stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1776,7 +1777,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
             defer { _idx += 1 }
             let bufferSize = MemoryLayout.size(ofValue: _buffer)
             return withUnsafeMutablePointer(to: &_buffer) { ptr_ in
-                let ptr = UnsafeMutableRawPointer(ptr_).assumingMemoryBound(to: UInt8.self)
+                let ptr = UnsafeMutableRawPointer(ptr_).bindMemory(to: UInt8.self, capacity: bufferSize)
```

Comment:
This is the most interesting case. We definitely don't want to use `assumingMemoryBound` here. There's just no way to ensure that's correct. We should be able to iterate over Data, returning each `UInt8` without actually binding the memory, just like we do with `UnsafeRawBufferPointer`. Clearly, there needs to be a `Data.copyBytes(to: UnsafeMutableRawPointer)`. It would be trivial to add. That actually should be part of the public API additions/changes that we discussed a couple years ago, but maybe the one that takes `NSRange` is sufficient?

Reply:
This particular case is not the data's bytes but instead a buffer in the iterator so it is fine. `_buffer` (being the source of `ptr_`) is a tuple of `UInt8`.